pkg/sessionctx/variable/sysvar.go

```diff
@@ -3483,3 +3490,109 @@ func setTiFlashComputeDispatchPolicy(s *SessionVars, val string) error {
 	s.TiFlashComputeDispatchPolicy = p
 	return nil
 }
+
+func setPipelinedDmlResourcePolicy(s *SessionVars, val string) error {
+	// ensure the value is trimmed and lowercased
+	val = strings.TrimSpace(val)
+	lowVal := strings.ToLower(val)
+	switch lowVal {
+	case vardef.StrategyPerformance:
+		s.PipelinedDMLConfig.PipelinedFlushConcurrency = vardef.DefaultFlushConcurrency
+		s.PipelinedDMLConfig.PipelinedResolveLockConcurrency = vardef.DefaultResolveConcurrency
+		s.PipelinedDMLConfig.PipelinedWriteThrottleRatio = 0
+	case vardef.StrategyConservation:
+		s.PipelinedDMLConfig.PipelinedFlushConcurrency = vardef.ConservationFlushConcurrency
+		s.PipelinedDMLConfig.PipelinedResolveLockConcurrency = vardef.ConservationResolveConcurrency
+		s.PipelinedDMLConfig.PipelinedWriteThrottleRatio = 0
+	default:
+		// Create a temporary config to hold new values to avoid partial application
+		newConfig := PipelinedDMLConfig{
+			PipelinedFlushConcurrency:       vardef.DefaultFlushConcurrency,
+			PipelinedResolveLockConcurrency: vardef.DefaultResolveConcurrency,
+			PipelinedWriteThrottleRatio:     0,
+		}
```

Comment:
```suggestion
		newConfig := PipelinedDMLConfig{
			PipelinedFlushConcurrency:       s.PipelinedFlushConcurrency,
			PipelinedResolveLockConcurrency: s.PipelinedResolveLockConcurrency,
			PipelinedWriteThrottleRatio:     s.PipelinedWriteThrottleRatio,
		}
```

Shall we keep the previous configuration if the user don't set it?

Unless some variables will be reset when user update another one.

```sql
set @@tidb_pipelined_dml_resource_policy = 'custom{concurrency=64}' -- PipelinedFlushConcurrency is 64
set @@tidb_pipelined_dml_resource_policy = 'custom{write_throttle_ratio=0.5}' -- PipelinedFlushConcurrency is rest to 128
```

```

Reply:
It's good to just let the omitted fields be reset to the default value. Unless the actual variable may be different from `show variables like "tidb_pipelined_dml_resource_policy"`.